src/nupic/ntypes/ArrayBase.hpp

```diff
@@ -186,8 +238,8 @@ namespace nupic
   // Compare contents of two ArrayBase objects
   // Note: An Array and an ArrayRef could be the same if type, count, and buffer
   // contents are the same.
-  bool operator==(const ArrayBase &lhs, const ArrayBase &rhs);
-  inline bool operator!=(const ArrayBase &lhs, const ArrayBase &rhs) {return !(lhs == rhs);}
+  bool operator==(ArrayBase &lhs, ArrayBase &rhs);
+  inline bool operator!=(ArrayBase &lhs, ArrayBase &rhs) {return !(lhs == rhs);}
```

Comment:
Is this a mistake or intentional?  I though that you made `const ArrayBase` possible.

Reply:
>  I though that you made `const ArrayBase` possible.

Oh, good catch.  A left-over when I was trying to make ArrayBase work with non-const SDR.  I will put the const back in.